SnapSheet/Classes/SnapSheetViewController.swift

```diff
@@ -21,8 +21,14 @@ public class SnapSheetViewController: UIViewController {
 		}
 	}
 
+	public var didUpdateState: ((State) ->  (Void))?
+
 	/// The state of the sheet
-	public var state: State = .closed
+	public var state: State = .closed {
+		didSet {
+			didUpdateState?(state)
```

Comment:
PP: check current state is not the same as previous state
NTH: (not to do now: provide state transition: from to